ironfish/src/memPool/feeEstimator.ts

```diff
@@ -194,22 +207,34 @@ export class FeeEstimator {
 
     size += receives.length * NOTE_ENCRYPTED_SERIALIZED_SIZE_IN_BYTE
 
-    if (estimateFeeRate) {
-      const additionalAmountNeeded =
-        this.getFee(estimateFeeRate, size) - (amount - amountNeeded)
-
-      if (additionalAmountNeeded > 0) {
-        const { notesToSpend: additionalNotesToSpend } = await this.wallet.createSpends(
-          sender,
-          additionalAmountNeeded,
-        )
-        const additionalSpendsLength =
-          additionalNotesToSpend.length * SPEND_SERIALIZED_SIZE_IN_BYTE
-        size += additionalSpendsLength
-      }
-    }
+    const spenderChange = amount - amountNeeded
 
-    return size
+    return { size, spenderChange }
+  }
+
+  private getPendingTransactionFee(
```

Comment:
The logic here will be useful in calculating a fee from fee rate in `Wallet.createTransaction`